File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/states/teleop/examples/LiftTeleop.java
package org.firstinspires.ftc.teamcode.states.teleop.examples;

import org.firstinspires.ftc.teamcode.hardware.RobotHardware;
import org.firstinspires.ftc.teamcode.stateengine.State;

public class LiftTeleop implements State {

  RobotHardware rh = null;

  @Override
  public void init(RobotHardware rh) { this.rh = rh; }

  @Override
  public void run() {
    rh.exampleLiftH.setLiftPosition(rh.controls.exampleLiftSpeed());
  }

  @Override
  public void stop() {
    // TODO.
  }

  @Override
  public boolean isDone() { return false; }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/states/teleop/examples/LiftTeleop.kt
package org.firstinspires.ftc.teamcode.states.teleop.examples

import org.firstinspires.ftc.teamcode.hardware.RobotHardware
import org.firstinspires.ftc.teamcode.stateengine.State

class LiftTeleop() : State {
  private lateinit var rh: RobotHardware

  override fun init(rh: RobotHardware) {
    this.rh = rh
  }

  override fun run() {
    rh.exampleLiftH.setLiftPosition(rh.controls.exampleLiftSpeed())
  }

  override fun stop() {}

  override val isDone = false
}
